src/main/java/org/dependencytrack/policy/cel/CelPolicyQueryManager.java

```diff
@@ -144,9 +144,17 @@ List<ComponentProjection> fetchAllComponents(final long projectId, final Collect
                 )
                 .map(mapping -> "\"C\".\"%s\" AS \"%s\"".formatted(mapping.sqlColumnName(), mapping.javaFieldName()))
                 .collect(Collectors.joining(", "));
-
         final Query<?> query = pm.newQuery(Query.SQL, """
-                SELECT %s , "IMC"."PUBLISHED_AT" AS "publishedAt" FROM "COMPONENT" "C" LEFT JOIN "INTEGRITY_META_COMPONENT" "IMC" ON  "C"."PURL"="IMC"."PURL" WHERE "PROJECT_ID" = ?
+                SELECT %s , "IMC"."PUBLISHED_AT" as "publishedAt",
+                "RMC"."LATEST_VERSION" as "latestVersion"
+                from
+                "COMPONENT" "C"
+                LEFT JOIN "INTEGRITY_META_COMPONENT" "IMC" ON
+                "C"."PURL" = "IMC"."PURL"
+                LEFT JOIN "REPOSITORY_META_COMPONENT" "RMC" ON
+                "C"."NAME" = "RMC"."NAME"
```

Comment:
The selection of `publishedAt` and `latestVersion`, and consequently the respective `JOIN`s with `INTEGRITY_META_COMPONENT` and `REPOSITORY_META_COMPONENT` should *only* be done when those fields are required by a script. Otherwise those `JOIN`s will only impact performance negatively.